hazelcast/src/main/java/com/hazelcast/jet/core/JetProperties.java

```diff
@@ -135,6 +135,48 @@
     public static final HazelcastProperty JET_IDLE_COOPERATIVE_MAX_MICROSECONDS
         = new HazelcastProperty("jet.idle.cooperative.max.microseconds", 500, MICROSECONDS);
 
+    /**
+     * The minimum time in milliseconds the cooperative worker threads
+     * will sleep if there is no tasklet to run. Lower values increase
+     * idle CPU usage but may result in decreased latency when starting
+     * the execution of a job. Higher values may add a delay to the
+     * starting the execution of a job.
+     * <p>
+     * The default value is {@code 1ms}.
+     * <p>
+     * Note: the underlying {@link LockSupport#parkNanos(long)} call may
+     * actually sleep longer depending on the operating system (up to 15000µs on
+     * Windows). See the <a
+     * href="https://hazelcast.com/blog/locksupport-parknanos-under-the-hood-and-the-curious-case-of-parking/">
+     * Hazelcast blog post about this subject</a> for more details.
+     * <p>
+     * See also: {@link #JET_IDLE_COOPERATIVE_MIN_MICROSECONDS}, {@link #JET_IDLE_COOPERATIVE_MAX_MICROSECONDS},
+     * {@link #JET_IDLE_COOPERATIVE_NO_TASKLET_MAX_MILLISECONDS}
+     */
+    public static final HazelcastProperty JET_IDLE_COOPERATIVE_NO_TASKLET_MIN_MILLISECONDS
```

Comment:
Thinking about this, we actually wake up cooperative workers when a new job is submitted: https://github.com/hazelcast/hazelcast/blob/c17682d96cbdc65fe92b0b268e6e6e7f38a4ea35/hazelcast/src/main/java/com/hazelcast/jet/impl/execution/TaskletExecutionService.java#L206

Maybe we need to improve this mechanism and sleep for 1 year if there's no tasklet to run and rely on this. In that case we could also start the threads immediately and we won't need this property.

Reply:
This approach assumes that once a thread has no tasklets, then it shouldn't receive any tasklet during that jobs life time. Is this ok? Is it guaranteed that in one job a thread will never be without a tasklet, or even if it has no tasklets then that thread won't be needed until that job ends?

Reply:
Once a thread has no tasklet, new tasklet can be added only through `submitCooperativeTasklets`. And that method always unparks all workers.

Reply:
I've thought about this too but I think there is a race: 
- coop-thread checks the tasklets and it is empty
- a new job is submitted which inserts the tasklets and calls unpark
- coop-thread parks

This can happen currenly too, so starting of a job can be delayed at most `JET_IDLE_COOPERATIVE_NO_TASKLET_MAX_MILLISECONDS`

Reply:
@gurbuzali good point, you're right. I'm not sure if we can fix that somehow...

Reply:
I wrote ec0eac7 in order to fix this. I pushed this commit because I think it's easier to look at here when pushed. Please feel free to review it or change in case a better replacement is found. If there are problems then we also can drop this commit. However if not then this commit should alone enough to solve idle spinning.